backend/internal/utils/query_params/pagination.go

```diff
@@ -0,0 +1,29 @@
+package query_params
+
+import (
+	"github.com/gofiber/fiber/v2"
+	"strconv"
+)
+
+func GetPaginationParams(c *fiber.Ctx) (int, int, bool, error) {
+	offsetParam := c.Query("offset", "0")
+	limitParam := c.Query("limit", "10")
+	fetchAll := c.Query("fetch_all", "false")
```

Comment:
The only thing here is I want the frontend to be specific about what they want rather than just giving them all. I agree with you on the specific /leagues/{id}/games will likely be how we are fetching anyway.